ComponentKit/Core/Scope/CKComponentScopeRootFactory.mm

```diff
@@ -0,0 +1,25 @@
+//
+//  CKComponentScopeRootFactory.m
+//  ComponentKit
+//
+//  Created by Oliver Rickard on 3/28/17.
+//
+//
+
+#import "CKComponentScopeRootFactory.h"
+
+#import "CKComponentControllerAppearanceEvents.h"
+#import "CKComponentBoundsAnimationPredicates.h"
+
+CKComponentScopeRoot *CKComponentScopeRootWithListener(id<CKComponentStateListener> listener)
+{
+  return [CKComponentScopeRoot
+   rootWithListener:listener
+   componentPredicates:{
+     &CKComponentBoundsAnimationPredicate
+   }
+   componentControllerPredicates:{
+     &CKComponentControllerAppearanceEventPredicate,
+     &CKComponentControllerDisappearanceEventPredicate
+   }];
```

Comment:
Oh to be clear, this is the efficient traversal. We build up the list of matching components on component generation, so by nature this is cached. It happens once for a scope root, and then the results are used over the lifetime.